rs/registry/canister/src/mutations/do_create_subnet.rs

```diff
@@ -840,63 +736,23 @@ mod test {
         registry.maybe_apply_mutation_internal(fake_subnet_mutation);
 
         // Step 2: Try to create another subnet with duplicate keys, which should panic.
-        let key_request = EcdsaKeyRequest {
-            key_id,
-            subnet_id: Some(*TEST_USER1_PRINCIPAL),
-        };
-        let payload = CreateSubnetPayload {
-            replica_version_id: ReplicaVersion::default().into(),
-            ecdsa_config: Some(EcdsaInitialConfig {
-                quadruples_to_create_in_advance: 1,
-                keys: vec![key_request; 2],
+        let key_config_request = KeyConfigRequest {
+            key_config: Some(KeyConfig {
+                key_id: Some(MasterPublicKeyId::Ecdsa(key_id)),
+                pre_signatures_to_create_in_advance: Some(1),
                 max_queue_size: Some(DEFAULT_ECDSA_MAX_QUEUE_SIZE),
-                signature_request_timeout_ns: None,
-                idkg_key_rotation_period_ms: None,
             }),
-            ..Default::default()
-        };
-        futures::executor::block_on(registry.do_create_subnet(payload));
-    }
-
-    // TODO[NNS1-3022]: Replace this test with one that checks that `subnet_record.ecdsa_config`
```

Comment:
Quick question: It mentions here, that this should be replace with a test that `subnet_record.ecdsa_config` cannot be set.

Do we have such a test? IIUC, we are simply removing this test right now.

Reply:
You're right that I didn't add tests to check that setting legacy ECDSA config is now forbidden.

That should be okay, as in a follow-up PR I'm going to remove that (optional, obsolete) field entirely.